0004/README.md

```diff
@@ -0,0 +1,524 @@
+
+*   A package version *satisfies* a provenance policy if its provenance attestation properties meet the policy expectations for the package name.
+
+## Objectives
+
+Evaluate policies over SLSA provenance for npm software packages.
+
+Non-objectives:
+
+-  Policies for what direct or indirect package dependencies (or dependency
+    versions) to expect for an npm software package; policies for vulnerability
+    scans.
+
+## Overview
+
+Here, we highlight the main architecture elements, their roles, and limitations.
+We present more detail in subsequent sections, with additional detail in the
+Appendix.
+
+### Extended software supply chain model
+
+![Extended model](extended-model.svg)
+
+The figure shows a generic supply chain for software packages, from source repository
+to installation, extended with two SLSA provenance policy evaluation points:
+
+-  When a software package is published to a package registry.
+-  When a software package is installed from a package registry.
+
+A SLSA provenance policy contains expectations for provenance, as discussed in later
+sections.
+
+SLSA Provenance expectations may be evaluated at other points inside or outside
+the above pipeline. Examples are a pre-build provenance check for build inputs,
+or an organization-wide proxy that checks SLSA provenance for package downloads.
+SLSA Provenance expectations may also differ between development versus
+production environments, or private versus public registries. We will not cover
+these differences in detail, but expect that the use cases can use similar SLSA
+provenance expectations as described here.
+
+### SLSA provenance expectations, benefits and limitations
+
+Each SLSA provenance policy covers multiple versions for a specific package
+name. For example, the policy for package ‘foo' could expect that each covered
+version:
+
+-  Has SLSA provenance that is signed with a key that is bound to a trusted
+    identity. The key could be long-lived, rotated regularly, or ephemeral, and
+    the identity could be a builder, individual, or organization.
+-  The provenance contains claims that:
+    -  Combined together satisfy the requirements for SLSA level X.
+        -  A policy evaluator may limit the maximum SLSA level
+            depending on the builder identity or other context. 
+
+    -  The package was built from the repository
+        [https://example.com/path/to/foo](https://example.com/path/to/foo).
+
+-  Other provenance claims may be relevant for specific use cases.
+
+When enforced, these expectations will ensure that a compromise of a package
+registry upload credential (or a compromise of the registry itself) cannot
+result in the installation of a rogue ‘foo' package version that was built from
+an unexpected repository or with an unexpected builder. 
+
+If a ‘good' package version does not meet policy expectations, the expectations
```

Comment:
Sorry, but how? Because now you have two problems instead of one: updating packages _and_ policies securely. Shouldn't a policy itself be versioned? If so, wouldn't you associate different policies for different ranges of versions of packages?

Reply:
There is policy and there is distribution. Without policy there is no distribution, and this doc is primarily concerned with policy.

Provenance policy expectations need to support package versions (ranges). This is now explicit in 'terminology' and a few other places in the text. I can add more if desirable.

The TOFU approach is for bootstrapping in the absence of a secure policy management and distribution system.  It allows us to learn what works, policy-wise.

Reply:
Makes sense for prototyping purposes, as @MarkLodato explained earlier today. Could we please mark distribution as out of scope for this proposal?

Reply:
Added in the non-objectives section under  'policy distribution'